services/121-service/src/transaction-jobs/processors/transaction-jobs-intersolve-voucher.processor.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { TestBed } from '@automock/jest';
+import { Job } from 'bull';
+import Redis from 'ioredis';
+
+import { REDIS_CLIENT } from '@121-service/src/payments/redis/redis-client';
+import { TransactionJobsProcessorIntersolveVoucher } from '@121-service/src/transaction-jobs/processors/transaction-jobs-intersolve-voucher.processor';
+import { TransactionJobsIntersolveVoucherService } from '@121-service/src/transaction-jobs/services/transaction-jobs-intersolve-voucher.service';
+
+const mockPaymentJob = {
+  id: 11,
+  programId: 3,
+  userId: 1,
+  paymentNumber: 3,
+  referenceId: '40bde7dc-29a9-4af0-81ca-1c426dccdd29',
+  transactionAmount: 25,
+  isRetry: false,
+  bulkSize: 10,
+  name: 'mock-fail-create-debit-card',
+  addressStreet: 'Straat',
+  addressHouseNumber: '1',
+  addressHouseNumberAddition: 'A',
+  addressPostalCode: '1234AB',
+  addressCity: 'Den Haag',
+  phoneNumber: '14155238886',
+};
+const testJob = { data: mockPaymentJob } as Job;
+
+describe('Payment processor(s)', () => {
+  let transactionJobsIntersolveVoucherService: jest.Mocked<TransactionJobsIntersolveVoucherService>;
+  let paymentProcessor: TransactionJobsProcessorIntersolveVoucher;
+  let redisClient: jest.Mocked<Redis>;
+
+  beforeAll(() => {
+    const { unit, unitRef } = TestBed.create(
+      TransactionJobsProcessorIntersolveVoucher,
+    )
+      .mock(TransactionJobsIntersolveVoucherService)
+      .using(transactionJobsIntersolveVoucherService)
+      .mock(REDIS_CLIENT)
+      .using(redisClient)
+      .compile();
+
+    paymentProcessor = unit;
+    transactionJobsIntersolveVoucherService = unitRef.get(
+      TransactionJobsIntersolveVoucherService,
+    );
+    redisClient = unitRef.get(REDIS_CLIENT);
+  });
+
+  it('should call processIntersolveVoucherTransactionJob', async () => {
```

Comment:
Now that I reread it I don't think the processor.spec.ts files add a lot of value. What do you think?

Reply:
I agree. It's there all for all FSPs, but the functionality is indeed completely the same for all FSPs, so no need to test them all separately. Conversely, in the transaction-queue.service.spec, we also don't have a test for each FSP because the functionality is the same. Shall I remove it for all but one?